cognite/extractorutils/uploader.py

```diff
@@ -1166,3 +1173,134 @@ def __len__(self) -> int:
             Number of events in queue
         """
         return self.upload_queue_size
+
+
+class BytesUploadQueue(AbstractUploadQueue):
+    def __init__(
+        self,
+        cdf_client: CogniteClient,
+        post_upload_function: Optional[Callable[[List[Any]], None]] = None,
+        max_queue_size: Optional[int] = None,
+        max_upload_interval: Optional[int] = None,
+        trigger_log_level: str = "DEBUG",
+        thread_name: Optional[str] = None,
+        overwrite_existing: bool = False,
+        cancelation_token: threading.Event = threading.Event(),
+    ):
+        super().__init__(
+            cdf_client,
+            post_upload_function,
+            max_queue_size,
+            max_upload_interval,
+            trigger_log_level,
+            thread_name,
+            cancelation_token,
+        )
+        self.upload_queue: List[Tuple[bytes, FileMetadata]] = []
+        self.overwrite_existing = overwrite_existing
+        self.upload_queue_size = 0
+        self.latency_zero_point = arrow.utcnow()
+
+        # Prometheus
+        self.bytes_queued = BYTES_UPLOADER_QUEUED
+        self.queue_size = BYTES_UPLOADER_QUEUE_SIZE
+        self.latency = BYTES_UPLOADER_LATENCY
+        self.bytes_written = BYTES_UPLOADER_WRITTEN
+
+    def add_to_upload_queue(self, frame: bytes, metadata: FileMetadata) -> None:
+        """
+        Add the frame to the upload queue
+        :param frame: bytes
+        :param metadata:
+        :return:
+        """
+        with self.lock:
+            if self.upload_queue_size == 0:
+                self.latency_zero_point = arrow.utcnow()
+
+            self.upload_queue.append((frame, metadata))
+            self.upload_queue_size += 1
+            self.bytes_queued.inc()
+            self.queue_size.set(self.upload_queue_size)
+
+    def upload(self) -> None:
```

Comment:
I wonder if we could generalize and reuse some parts of the fileuploader to avoid duplicating code for the upload part, but I suggest we do a separate refactoring of that later

Reply:
I was thinking about that same, but agree, we can refactor that later 💪🏻